### DISCUSSION TITLE
`from __future__ import annotations` leads to different openapi schema with return type None

### DISCUSSION THREAD

**Discussion by fnep** [fastapi/fastapi#9723] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from __future__ import annotations

import json
from sys import stdout

from fastapi import FastAPI

app = FastAPI()

@app.post("/")
def hello_world() -> None:
    pass

json.dump(app.openapi(), indent=2, fp=stdout)
```

### Description

With `from __future__ import annotations` in the code, this returns:
```json
{
  "openapi": "3.0.2",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "post": {
        "summary": "Hello World",
        "operationId": "hello_world__post",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "title": "Response Hello World  Post",
                  "type": "null"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

Without `from __future__ import annotations`, it returns:
```json
{
  "openapi": "3.0.2",
  "info": {
    "title": "FastAPI",
    "version": "0.1.0"
  },
  "paths": {
    "/": {
      "post": {
        "summary": "Hello World",
        "operationId": "hello_world__post",
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {}
              }
            }
          }
        }
      }
    }
  }
}
```

I think both should generate the same OpenAPI Schema, but `paths./.post.responses.200.content.application/json.schema` is different. We notived this, because we ran the result against openapi-spec-validator and the version with `from __future__ import annotations` is marked as invalid.

### Operating System

macOS

### FastAPI Version

0.98.0

### Python Version

3.10.3

**Comment by arjwilliams** [comment, marked as answer]
Issue is with the annotation import transforming the return type annotation to a string `"None"`, which is different from the return type annotation when the annotation import is not applied, which is just `None` (non-string).

FastAPI was then treating the non-string `None` value version the same as having no return value annotation, but treating the string `"None"` value version as still having a return annotation.

This issue should be fixed by #9425.

**Reply by fnep** [reply]
Thank you for pointing me to this issue. I tested the branch and it fixes the issue.
